Sources/SolanaSwift/Programs/TokenProgram.swift

```diff
@@ -0,0 +1,288 @@
+import Foundation
+
+public protocol TokenProgram {
+    func initializeMintInstruction(
+        mint: PublicKey,
+        decimals: UInt8,
+        authority: PublicKey,
+        freezeAuthority: PublicKey?
+    ) -> TransactionInstruction
+    
+    func initializeAccountInstruction(
+        account: PublicKey,
+        mint: PublicKey,
+        owner: PublicKey
+    ) -> TransactionInstruction
+    
+    func transferInstruction(
+        source: PublicKey,
+        destination: PublicKey,
+        owner: PublicKey,
+        amount: UInt64
+    ) -> TransactionInstruction
+    
+    func transferCheckedInstruction(
+        source: PublicKey,
+        mint: PublicKey,
+        destination: PublicKey,
+        owner: PublicKey,
+        multiSigners: [Account],
+        amount: Lamports,
+        decimals: Decimals
+    ) -> TransactionInstruction
+    
+    func burnCheckedInstruction(
+        mint: PublicKey,
+        account: PublicKey,
+        owner: PublicKey,
+        amount: UInt64,
+        decimals: UInt8
+    ) -> TransactionInstruction
+    
+    func approveInstruction(
+        account: PublicKey,
+        delegate: PublicKey,
+        owner: PublicKey,
+        multiSigners: [Account],
+        amount: UInt64
+    ) -> TransactionInstruction
+    
+    func mintToInstruction(
+        mint: PublicKey,
+        destination: PublicKey,
+        authority: PublicKey,
+        amount: UInt64
+    ) -> TransactionInstruction
+    
+    func closeAccountInstruction(
+        account: PublicKey,
+        destination: PublicKey,
+        owner: PublicKey
+    ) -> TransactionInstruction
+    
+    func closeAccountInstruction(
+        account: PublicKey,
+        destination: PublicKey,
+        owner: PublicKey,
+        signers: [PublicKey]
+    ) -> TransactionInstruction
+}
+
+public struct TokenProgramImpl: SolanaBasicProgram, TokenProgram {
```

Comment:
Can we rename it somehow? `TokenProgramImpl` looks not really nice

Reply:
DefaultTokenProgram is also nice